### PR TITLE
Update CI pipelines

### PR DESCRIPTION
**Pull request by guizmaii**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -10,57 +10,56 @@ on:
...
+        java: [ '8', '11', '17' ]
+        scala: [ '2.12.15', '2.13.8', '3.1.1' ]
     steps:
-      - name: Checkout current branch
-        uses: actions/checkout@v2.3.4
+      - uses: actions/checkout@v2
         with:
           fetch-depth: 0
-      - name: Setup Scala and Java
-        uses: olafurpg/setup-scala@v13
```
Comment (guizmaii) [inline .github/workflows/ci.yml:43]: `olafurpg/setup-scala` is deprecated and `sbt` is now installed by default in all the OS proposed by Github Actions

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -10,57 +10,56 @@ on:
...
       - name: Mdocs check
         run: sbt docs/mdoc
 
   test:
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-latest
     strategy:
       fail-fast: false
       matrix:
-        java: ['adopt@1.8', 'adopt@1.11']
```
Comment (guizmaii) [inline .github/workflows/ci.yml:35]: AdoptOpenJDK is deprecated and replaced by Eclipse Temurin

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -10,57 +10,56 @@ on:
...
+        scala: [ '2.12.15', '2.13.8', '3.1.1' ]
     steps:
-      - name: Checkout current branch
-        uses: actions/checkout@v2.3.4
+      - uses: actions/checkout@v2
         with:
           fetch-depth: 0
-      - name: Setup Scala and Java
-        uses: olafurpg/setup-scala@v13
+      - uses: coursier/cache-action@v6
```
Comment (svroonland) [inline .github/workflows/ci.yml:40]: Can we keep the action names?

**Review by svroonland — CHANGES_REQUESTED** [review]

File: .github/workflows/auto-approve.yml [hunk 4]
```diff
@@ -5,7 +5,7 @@ on:
 
 jobs:
   auto-approve:
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-latest
```
[the remark this answers is not in the document]
Comment (guizmaii) [reply .github/workflows/auto-approve.yml:8]: Up to you. IMO, we're on the JVM, that doesn't matter much.

[the remark this answers is not in the document]
Comment (guizmaii) [reply .github/workflows/auto-approve.yml]: Done!

File: .github/workflows/ci.yml (hunk 3 above)
Comment (guizmaii) [reply .github/workflows/ci.yml]: Done!

**Review by svroonland — APPROVED** [review]

**Pull request merged**